File: LeetCode/April Challenge/25th Day/Day_25.cpp
class Solution {

public:
        Solution()
        {
            ios::sync_with_stdio(false); 
            std::cin.tie(nullptr); 
            std::cout.tie(nullptr);
        }
    bool canJump(vector<int>& nums) {
        int maxjump=0;
        for(int i=0;i<nums.size();i++)
        {
            if(i>maxjump)
                return false;
            maxjump=max(i+nums[i], maxjump);
        }
        return true; 
    }
};

File: LeetCode/April Challenge/25th Day/Day_25.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if not nums:
            return False
        n = len(nums)
        target = n - 1
        for i in range(n - 2, -1, -1):
            if nums[i] + i >= target:
                target = i
        return not target
        
